hardware_pydantic/devices.py

```diff
@@ -56,26 +88,97 @@ def action__transfer(
             self,
             from_obj: Rack | Heater,
```

Comment:
If the move operation has to specify a `vial`, why do we need `from_obj`? Shouldn't we just have to say the vial and its destination, since the vial knows where it is currently located?

Reply:
I didn't think much about this, in this case specifying `vial` should be enough. 

Now I think about this... there is a difference between
1. transfer the `vial` on `rack1.16A` to somewhere
2. transfer `vial2`, currently located at `rack1.16A` to somewhere

Idk which is more appropriate, but both should work since  `vial` and `rack` know their parent and children, respectively.

Reply:
I think that in composing the action DAG, we shouldn't need to keep track of where a vial happens to be. An extreme example of this would be pulling a vial out of a chemical inventory system -- why should we need to specify where a specific vial containing a specific compound is currently located?

Reply:
I agree that most of the time we are moving a specific vial, not a vial that happens to be at position X. The only situation I can think of is for some reason we want to "free" the position occupied by something, so the instruction would be "move whatever at 16C to position X".

Reply:
For now, we can probably neglect that situation. I agree though -- for example, if a vial is sitting (done) after using the V-10 evaporator, nothing else will be able to use it until it gets out of the way. The simulator should help us identify such situations and understand why that vial was abandoned there